src/plugins/intel_cpu/tests/functional/custom/single_layer_tests/slice_scatter.cpp

```diff
@@ -83,7 +83,13 @@ class SliceScatterLayerCPUTest : public testing::WithParamInterface<SliceScatter
                                                                  in_data);
             } else {
                 // Fill the slice input2~input5 with specified data.
-                tensor = ov::Tensor{ov::element::i64, targetInputStaticShapes[i], inputValues[i - 2]};
+                auto inputValue = inputValues[i - 2];
+                if (!inputValue) {
+                    const auto param = ov::as_type_ptr<const ov::op::v0::Parameter>(funcInput.get_node_shared_ptr());
+                    tensor = ov::Tensor{ov::element::i64, targetInputStaticShapes[i]};
+                } else {
+                    tensor = ov::Tensor{ov::element::i64, targetInputStaticShapes[i], inputValue};
```

Comment:
Actually this is the reason why the tests don't cover the main issue, as is being resolved in this PR. The thing is that the SL tests in the static shapes configuration perform only one inference, thus the values on the variable inputs don't really change from infer to infer and SL tests don't spot the issue. But if we had had such tests, they would have revealed the design flaw of the current solution solution (please see the comment regarding `execute` method implementation). So apparently the SL tests should be extended to cover such a use case: the values on the variable inputs actually changes each inference. To this end even a dedicated test class may be introduced, when necessary.